From db84b021f80d3bf1b3dc53efa5b79bc0edcd5759 Mon Sep 17 00:00:00 2001
From: rivena port <port@localhost>
Date: Mon, 26 Oct 2026 08:23:30 +0800
Subject: [PATCH 2/8] sm8250-common: Switch to the legacy libion implementation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Android 17 的 libion 拆成了可配置实现，老 vendor blob（ION ioctl）
必须显式选 legacy_impl，并引入对应的 sepolicy，否则 ion 相关
neverallow 会直接卡住编译。
---
 BoardConfigCommon.mk | 1 +
 common.mk            | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index b1fe1cc..bc9db81 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -137,6 +137,7 @@ ENABLE_VENDOR_RIL_SERVICE := true
 VENDOR_SECURITY_PATCH := 2025-08-01
 
 # SEPolicy
+include device/lineage/sepolicy/libion/sepolicy.mk
 include device/qcom/sepolicy_vndr/SEPolicy.mk
 include hardware/oplus/sepolicy/qti/SEPolicy.mk
 
diff --git a/common.mk b/common.mk
index 5f4a4b8..bb5ccc6 100644
--- a/common.mk
+++ b/common.mk
@@ -168,6 +168,9 @@ PRODUCT_PACKAGES += \
 PRODUCT_COPY_FILES += \
     $(LOCAL_PATH)/configs/privapp-permissions-hotword.xml:$(TARGET_COPY_OUT_PRODUCT)/etc/permissions/privapp-permissions-hotword.xml
 
+# ION
+$(call soong_config_set_bool,libion,legacy_impl,true)
+
 # IPACM
 PRODUCT_PACKAGES += \
     ipacm \
-- 
2.43.0

